### ISSUE TITLE
Kotlin Android Extensions & Butterknife are deprecated

### ISSUE THREAD

**Issue by psh** [commons-app/apps-android-commons#4664]
**Summary:** 

Looking at "The future of Kotlin Android Extensions" from 23 November 2020, (see: https://android-developers.googleblog.com/2020/11/the-future-of-kotlin-android-extensions.html)

> The deprecation period starts with Kotlin 1.4.20, released today. android-kotlin-extensions will be removed in a future Kotlin release during or after September 2021. Long term, we will continue to maintain the kotlin-parcelize plugin, and you can continue to file issues on Parcelize in the Android Studio issue tracker.

According to Google (https://developer.android.com/topic/libraries/view-binding/migration)
> Kotlin Android Extensions is deprecated, which means that using Kotlin synthetics for view binding is no longer supported. If your app uses Kotlin synthetics for view binding, use this guide to migrate to Jetpack view binding.

In addition, the author of Butternife has stated (https://github.com/JakeWharton/butterknife)
> *Attention:* This tool is now deprecated. Please switch to view binding. Existing versions will continue to work, obviously, but only critical bug fixes for integration with AGP will be considered. Feature development and general bug fixes have stopped.

**Would you like to work on the issue?**

It looks like a pretty big migration, with a large number of files that will be touched.  A search for `import butterknife.BindView` shows 40 files, and `import kotlinx.android.synthetic` a further 16 files; probably too big for a single person and something to do in stages.  I would be happy to be part of the process.

**Comment by misaochan** [comment]
Sounds good, thanks @madhurgupta10 . Let's do it. :) We can probably split this up by package - 1 PR for files in 1 package?

**Comment by madhurgupta10** [comment]
@misaochan Yes sure, even more than one PR per package would be okay IMO.

I will create a PR with the necessary dependencies and an example so it is easier for first-time contributors to follow up :)

**Comment by ThomasLatham** [comment]
Hi, I'd really like to help out with this migration as my first open-source contribution. 

I've followed the quick start guide for developers (in addition to reading the Contributing Guidelines, Developer workflow and Code Style), and Gradle has built the application successfully. Android Studio is still telling me there are errors, however, and I want to clear those up before I try writing any code, even seemingly straight-forward code like in the example provided by @madhurgupta10.

I read at the bottom of the quick start guide that I should open a new issue for trouble with setup, but I don't feel like I have enough experience to determine whether the problems I'm facing deserve their own issue. This probably isn't the right place to say all this either, but I've been stuck for a few days, and I just really want to be able to contribute.

Anyway, would somebody please be willing to help me out?

**Comment by madhurgupta10** [comment]
@ThomasLatham Feel free to open a new issue if you need help, one or more team members will be happy to help you out :)

**Comment by psh** [comment]
@ThomasLatham if it helps - my process for the last few PR that I submitted was to search for `import kotlinx.android.synthetic` and choose a file from the list that isn't a base class of a hierarchy; right now there are a half-dozen or so files with synthetics left.  Personally, I like to be _finishing_ rather than _starting_, and since synthetics are closer to being finished, that would be where I'd jump in so we can cross that off the list before starting Butterknife.

**Comment by ThomasLatham** [comment]
Thank you @madhurgupta10! I'll see what I can figure out on my own, but I won't hesitate to open a new issue if I get stuck

**Comment by psh** [comment]
https://android-developers.googleblog.com/2022/02/discontinuing-kotlin-synthetics-for-views.html?m=1

Google says -

> In November 2020, we announced that this plugin has been deprecated in favor of better solutions, and we recommended removing the plugin from your projects. We know many developers still depend on this plugin’s features, and we’ve extended the support timeframe so you have more time to complete your migrations.
> 
> We are now setting a deadline for these migrations: the plugin will be removed in Kotlin 1.8, which is expected to be released by the end of 2022. At that time, you won’t be able to update your project to newer Kotlin versions if it still depends on the Kotlin Android Extensions plugin. This means that now is the time to perform the necessary migrations in your projects.

**Comment by bosankus** [comment]
Hi All @psh @misaochan @madhurgupta10 @ThomasLatham @prtksxna ,
When I search `import butterknife.BindView` globally in Android Studio, I see multiple classes where its still being used. And   out of which, I want to start working for the mentioned change at `WelcomeActivity`.

Can I raise a PR for it ?

**Comment by bosankus** [comment]
Hi @nicolas-raoul , Can you review the above PRs kindly?

**Comment by glitched-sudhanshu** [comment]
Hi @madhurgupta10 is this issue is solved.
What I understood from all this discussion was, it needs multiple people.
So can I work into it?

**Comment by nicolas-raoul** [comment]
@bosankus  Sorry for the delay, I have started reviewing but not finished yet.

@glitched-sudhanshu If you find other classes that need the same, please list them here and you can start working on them 🙂

**Comment by bosankus** [comment]
Hi @nicolas-raoul , there are other areas to cover too which am yet to push code in future for closing this ticket. Are you sure this issue can be closed now ?

**Comment by nicolas-raoul** [comment]
Thanks @bosankus!

You are right, still 10+42 to go 😅
Anyone feel free to join the effort, see the example PR https://github.com/commons-app/apps-android-commons/pull/4681/files 🙂
```
$ grep -R "import kotlinx.android.synthetic" app/src
app/src/test/kotlin/fr/free/nrw/commons/recentlanguages/RecentLanguagesAdapterUnitTest.kt:import kotlinx.android.synthetic.main.row_item_languages_spinner.view.*
app/src/main/java/fr/free/nrw/commons/nearby/fragments/AdvanceQueryFragment.kt:import kotlinx.android.synthetic.main.fragment_advance_query.*
app/src/main/java/fr/free/nrw/commons/explore/paging/BasePagingFragment.kt:import kotlinx.android.synthetic.main.fragment_search_paginated.*
app/src/main/java/fr/free/nrw/commons/explore/paging/FooterAdapter.kt:import kotlinx.android.synthetic.main.list_item_load_more.*
app/src/main/java/fr/free/nrw/commons/explore/media/PageableMediaFragment.kt:import kotlinx.android.synthetic.main.fragment_search_paginated.*
app/src/main/java/fr/free/nrw/commons/explore/media/PagedMediaAdapter.kt:import kotlinx.android.synthetic.main.layout_category_images.*
app/src/main/java/fr/free/nrw/commons/recentlanguages/RecentLanguagesAdapter.kt:import kotlinx.android.synthetic.main.row_item_languages_spinner.view.*
app/src/main/java/fr/free/nrw/commons/customselector/ui/selector/FolderFragment.kt:import kotlinx.android.synthetic.main.fragment_custom_selector.*
app/src/main/java/fr/free/nrw/commons/customselector/ui/selector/FolderFragment.kt:import kotlinx.android.synthetic.main.fragment_custom_selector.view.*
app/src/main/java/fr/free/nrw/commons/upload/LanguagesAdapter.kt:import kotlinx.android.synthetic.main.row_item_languages_spinner.view.*
```
```
$ grep -R "import butterknife.BindView" app/src
app/src/test/kotlin/fr/free/nrw/commons/review/ReviewActivityTest.kt:import butterknife.BindView
app/src/main/java/fr/free/nrw/commons/settings/SettingsActivity.java:import butterknife.BindView;
app/src/main/java/fr/free/nrw/commons/category/CategoryDetailsActivity.java:import butterknife.BindView;
app/src/main/java/fr/free/nrw/commons/contributions/MainActivity.java:import butterknife.BindView;
app/src/main/java/fr/free/nrw/commons/contributions/ContributionsListFragment.java:import butterknife.BindView;
app/src/main/java/fr/free/nrw/commons/contributions/ContributionsFragment.java:import butterknife.BindView;
app/src/main/java/fr/free/nrw/commons/contributions/ContributionViewHolder.java:import butterknife.BindView;
app/src/main/java/fr/free/nrw/commons/auth/LoginActivity.java:import butterknife.BindView;
app/src/main/java/fr/free/nrw/commons/notification/NotificationActivity.java:import butterknife.BindView;
app/src/main/java/fr/free/nrw/commons/quiz/QuizActivity.java:import butterknife.BindView;
app/src/main/java/fr/free/nrw/commons/quiz/QuizResultActivity.java:import butterknife.BindView;
app/src/main/java/fr/free/nrw/commons/nearby/fragments/NearbyParentFragment.java:import butterknife.BindView;
app/src/main/java/fr/free/nrw/commons/explore/ExploreListRootFragment.java:import butterknife.BindView;
app/src/main/java/fr/free/nrw/commons/explore/ExploreFragment.java:import butterknife.BindView;
app/src/main/java/fr/free/nrw/commons/explore/depictions/WikidataItemDetailsActivity.java:import butterknife.BindView;
app/src/main/java/fr/free/nrw/commons/explore/map/ExploreMapFragment.java:import butterknife.BindView;
app/src/main/java/fr/free/nrw/commons/explore/ExploreMapRootFragment.java:import butterknife.BindView;
app/src/main/java/fr/free/nrw/commons/explore/recentsearches/RecentSearchesFragment.java:import butterknife.BindView;
app/src/main/java/fr/free/nrw/commons/explore/SearchActivity.java:import butterknife.BindView;
app/src/main/java/fr/free/nrw/commons/bookmarks/pictures/BookmarkPicturesFragment.java:import butterknife.BindView;
app/src/main/java/fr/free/nrw/commons/bookmarks/BookmarkFragment.java:import butterknife.BindView;
app/src/main/java/fr/free/nrw/commons/bookmarks/BookmarkListRootFragment.java:import butterknife.BindView;
app/src/main/java/fr/free/nrw/commons/bookmarks/locations/BookmarkLocationsFragment.java:import butterknife.BindView;
app/src/main/java/fr/free/nrw/commons/bookmarks/items/BookmarkItemsFragment.java:import butterknife.BindView;
app/src/main/java/fr/free/nrw/commons/profile/achievements/AchievementsFragment.java:import butterknife.BindView;
app/src/main/java/fr/free/nrw/commons/profile/leaderboard/LeaderboardFragment.java:import butterknife.BindView;
app/src/main/java/fr/free/nrw/commons/profile/ProfileActivity.java:import butterknife.BindView;
app/src/main/java/fr/free/nrw/commons/review/ReviewActivity.java:import butterknife.BindView;
app/src/main/java/fr/free/nrw/commons/review/ReviewImageFragment.java:import butterknife.BindView;
app/src/main/java/fr/free/nrw/commons/navtab/MoreBottomSheetFragment.java:import butterknife.BindView;
app/src/main/java/fr/free/nrw/commons/campaigns/CampaignView.java:import butterknife.BindView;
app/src/main/java/fr/free/nrw/commons/upload/license/MediaLicenseFragment.java:import butterknife.BindView;
app/src/main/java/fr/free/nrw/commons/upload/UploadActivity.java:import butterknife.BindView;
app/src/main/java/fr/free/nrw/commons/upload/UploadMediaDetailAdapter.java:import butterknife.BindView;
app/src/main/java/fr/free/nrw/commons/upload/ThumbnailsAdapter.java:import butterknife.BindView;
app/src/main/java/fr/free/nrw/commons/upload/SimilarImageDialogFragment.java:import butterknife.BindView;
app/src/main/java/fr/free/nrw/commons/upload/mediaDetails/UploadMediaDetailFragment.java:import butterknife.BindView;
app/src/main/java/fr/free/nrw/commons/upload/categories/UploadCategoriesFragment.java:import butterknife.BindView;
app/src/main/java/fr/free/nrw/commons/upload/depicts/DepictsFragment.java:import butterknife.BindView;
app/src/main/java/fr/free/nrw/commons/media/MediaDetailFragment.java:import butterknife.BindView;
app/src/main/java/fr/free/nrw/commons/media/MediaDetailPagerFragment.java:import butterknife.BindView;
app/src/main/java/fr/free/nrw/commons/media/ZoomableActivity.kt:import butterknife.BindView
```

**Comment by nicolas-raoul** [comment]
If anyone knows a trick to prevent this issue from getting closed every time, please let me know... and thanks Paul for reopening when I forgot to! :-)

**Comment by psh** [comment]
Looking at Github documentation and feature requests, it seems you cant _disable_ the auto-closing feature and the suggested quick workaround is to avoid the keywords in PR comments that trigger it.  So, avoid saying "Fixes #4664" and maybe say something like "This PR relates to #4664" instead.  In theory that will generate a link without this issue being closed when the PR merges.

**Comment by psh** [comment]
Just checking in with this issue, with our overall progress and to share an interesting nugget of information from the Google Issue Tracker (from https://issuetracker.google.com/issues/173030256#comment10) where it says - 

> Databinding is in maintenance mode as well.
> We don't plan to support KSP nor recommend data binding usage at this stage since compose is our recommended UI solution.

also, 

> ViewBinding is only a build tool and does not run an annotation processor, hence it is not impacted by KAPT / KSP.
> 
> So if you are using Views, ViewBinding will keep working.

**Comment by nicolas-raoul** [comment]
Should we remove this section of the Developer Documentation which suggests the use of butterknife?
https://github.com/commons-app/commons-app-documentation/blob/master/android/Code-style.md#butterknife

**Comment by psh** [comment]
Good call!  Yes, that should be removed in favor of saying that the app ought to use Google's view binding; both the kotlin android extensions and the butterknife references should be removed.

**Comment by nicolas-raoul** [comment]
2024 update: 6+36 to go 🙂

```
grep -Rl "import kotlinx.android.synthetic" app/src
app/src/test/kotlin/fr/free/nrw/commons/recentlanguages/RecentLanguagesAdapterUnitTest.kt
app/src/test/kotlin/fr/free/nrw/commons/customselector/ui/selector/CustomSelectorActivityTest.kt
app/src/main/java/fr/free/nrw/commons/explore/paging/BasePagingFragment.kt
app/src/main/java/fr/free/nrw/commons/explore/paging/FooterAdapter.kt
app/src/main/java/fr/free/nrw/commons/explore/media/PagedMediaAdapter.kt
app/src/main/java/fr/free/nrw/commons/edit/EditActivity.kt
```

```
$ grep -Rl "import butterknife.BindView" app/src
app/src/test/kotlin/fr/free/nrw/commons/review/ReviewActivityTest.kt
app/src/main/java/fr/free/nrw/commons/settings/SettingsActivity.java
app/src/main/java/fr/free/nrw/commons/category/CategoryDetailsActivity.java
app/src/main/java/fr/free/nrw/commons/contributions/MainActivity.java
app/src/main/java/fr/free/nrw/commons/contributions/ContributionsFragment.java
app/src/main/java/fr/free/nrw/commons/contributions/ContributionViewHolder.java
app/src/main/java/fr/free/nrw/commons/notification/NotificationActivity.java
app/src/main/java/fr/free/nrw/commons/nearby/fragments/NearbyParentFragment.java
app/src/main/java/fr/free/nrw/commons/explore/ExploreListRootFragment.java
app/src/main/java/fr/free/nrw/commons/explore/ExploreFragment.java
app/src/main/java/fr/free/nrw/commons/explore/depictions/WikidataItemDetailsActivity.java
app/src/main/java/fr/free/nrw/commons/explore/map/ExploreMapFragment.java
app/src/main/java/fr/free/nrw/commons/explore/ExploreMapRootFragment.java
app/src/main/java/fr/free/nrw/commons/explore/recentsearches/RecentSearchesFragment.java
app/src/main/java/fr/free/nrw/commons/explore/SearchActivity.java
app/src/main/java/fr/free/nrw/commons/bookmarks/pictures/BookmarkPicturesFragment.java
app/src/main/java/fr/free/nrw/commons/bookmarks/BookmarkFragment.java
app/src/main/java/fr/free/nrw/commons/bookmarks/BookmarkListRootFragment.java
app/src/main/java/fr/free/nrw/commons/bookmarks/locations/BookmarkLocationsFragment.java
app/src/main/java/fr/free/nrw/commons/bookmarks/items/BookmarkItemsFragment.java
app/src/main/java/fr/free/nrw/commons/profile/leaderboard/LeaderboardFragment.java
app/src/main/java/fr/free/nrw/commons/profile/ProfileActivity.java
app/src/main/java/fr/free/nrw/commons/review/ReviewActivity.java
app/src/main/java/fr/free/nrw/commons/review/ReviewImageFragment.java
app/src/main/java/fr/free/nrw/commons/campaigns/CampaignView.java
app/src/main/java/fr/free/nrw/commons/upload/license/MediaLicenseFragment.java
app/src/main/java/fr/free/nrw/commons/upload/UploadActivity.java
app/src/main/java/fr/free/nrw/commons/upload/UploadMediaDetailAdapter.java
app/src/main/java/fr/free/nrw/commons/upload/ThumbnailsAdapter.java
app/src/main/java/fr/free/nrw/commons/upload/SimilarImageDialogFragment.java
app/src/main/java/fr/free/nrw/commons/upload/mediaDetails/UploadMediaDetailFragment.java
app/src/main/java/fr/free/nrw/commons/upload/categories/UploadCategoriesFragment.java
app/src/main/java/fr/free/nrw/commons/upload/depicts/DepictsFragment.java
app/src/main/java/fr/free/nrw/commons/media/MediaDetailFragment.java
app/src/main/java/fr/free/nrw/commons/media/MediaDetailPagerFragment.java
app/src/main/java/fr/free/nrw/commons/media/ZoomableActivity.kt
```

**Comment by ShashwatKedia** [comment]
> 2024 update: 6+36 to go 🙂

Since this issue is mentioned for GSoC, are we allowed to work on this now, or only the GSoC candidate will work on this later?

**Comment by nicolas-raoul** [comment]
@ShashwatKedia Yes, allowed, the Nearby tasks will fill in if this one is done already. 🙂

**Comment by neeldoshii** [comment]
Hi @nicolas-raoul can anyone work work on this issue (1package = 1Pr) at a time or you need to get assigned to this issue?

**Comment by shashankiitbhu** [comment]
Hi @nicolas-raoul , following are covered in Above PRs (Each PR addresses 1 Package , except last PR) :

app/src/test/kotlin/fr/free/nrw/commons/review/ReviewActivityTest.kt
app/src/main/java/fr/free/nrw/commons/settings/SettingsActivity.java

app/src/main/java/fr/free/nrw/commons/contributions/MainActivity.java
app/src/main/java/fr/free/nrw/commons/contributions/ContributionsFragment.java
app/src/main/java/fr/free/nrw/commons/contributions/ContributionViewHolder.java

app/src/main/java/fr/free/nrw/commons/explore/ExploreListRootFragment.java
app/src/main/java/fr/free/nrw/commons/explore/ExploreFragment.java
app/src/main/java/fr/free/nrw/commons/explore/depictions/WikidataItemDetailsActivity.java
app/src/main/java/fr/free/nrw/commons/explore/map/ExploreMapFragment.java
app/src/main/java/fr/free/nrw/commons/explore/ExploreMapRootFragment.java
app/src/main/java/fr/free/nrw/commons/explore/recentsearches/RecentSearchesFragment.java
app/src/main/java/fr/free/nrw/commons/explore/SearchActivity.java

app/src/main/java/fr/free/nrw/commons/bookmarks/pictures/BookmarkPicturesFragment.java
app/src/main/java/fr/free/nrw/commons/bookmarks/BookmarkFragment.java
app/src/main/java/fr/free/nrw/commons/bookmarks/BookmarkListRootFragment.java
app/src/main/java/fr/free/nrw/commons/bookmarks/locations/BookmarkLocationsFragment.java
app/src/main/java/fr/free/nrw/commons/bookmarks/items/BookmarkItemsFragment.java

app/src/main/java/fr/free/nrw/commons/profile/leaderboard/LeaderboardFragment.java
app/src/main/java/fr/free/nrw/commons/profile/ProfileActivity.java

app/src/main/java/fr/free/nrw/commons/upload/license/MediaLicenseFragment.java
app/src/main/java/fr/free/nrw/commons/upload/UploadActivity.java
app/src/main/java/fr/free/nrw/commons/upload/UploadMediaDetailAdapter.java
app/src/main/java/fr/free/nrw/commons/upload/ThumbnailsAdapter.java
app/src/main/java/fr/free/nrw/commons/upload/SimilarImageDialogFragment.java
app/src/main/java/fr/free/nrw/commons/upload/mediaDetails/UploadMediaDetailFragment.java
app/src/main/java/fr/free/nrw/commons/upload/categories/UploadCategoriesFragment.java
app/src/main/java/fr/free/nrw/commons/upload/depicts/DepictsFragment.java

**Comment by shashankiitbhu** [comment]
@neeldoshii I have already pushed PRs covering these too , it would great if you can chose Files that have not been covered yet (also please look at the list I have mentioned as covered , you can click on each of the PRs to check what files have been covered in each package), working on different PRs will help us complete this task effectively, thanks

**Comment by neeldoshii** [comment]
> Hi @nicolas-raoul , following are covered in the Above PRs (Each PR addresses 1 Package, except the last PR) :
> 
> app/src/main/java/fr/free/nrw/commons/category/CategoryDetailsActivity.java app/src/main/java/fr/free/nrw/commons/settings/SettingsActivity.java
> 
> app/src/main/java/fr/free/nrw/commons/contributions/MainActivity.java app/src/main/java/fr/free/nrw/commons/contributions/ContributionsFragment.java app/src/main/java/fr/free/nrw/commons/contributions/ContributionViewHolder.java
> 
> app/src/main/java/fr/free/nrw/commons/explore/ExploreListRootFragment.java app/src/main/java/fr/free/nrw/commons/explore/ExploreFragment.java app/src/main/java/fr/free/nrw/commons/explore/depictions/WikidataItemDetailsActivity.java app/src/main/java/fr/free/nrw/commons/explore/map/ExploreMapFragment.java app/src/main/java/fr/free/nrw/commons/explore/ExploreMapRootFragment.java app/src/main/java/fr/free/nrw/commons/explore/recentsearches/RecentSearchesFragment.java app/src/main/java/fr/free/nrw/commons/explore/SearchActivity.java
> 
> app/src/main/java/fr/free/nrw/commons/bookmarks/pictures/BookmarkPicturesFragment.java app/src/main/java/fr/free/nrw/commons/bookmarks/BookmarkFragment.java app/src/main/java/fr/free/nrw/commons/bookmarks/BookmarkListRootFragment.java app/src/main/java/fr/free/nrw/commons/bookmarks/locations/BookmarkLocationsFragment.java app/src/main/java/fr/free/nrw/commons/bookmarks/items/BookmarkItemsFragment.java
> 
> app/src/main/java/fr/free/nrw/commons/profile/leaderboard/LeaderboardFragment.java app/src/main/java/fr/free/nrw/commons/profile/ProfileActivity.java
> 
> app/src/main/java/fr/free/nrw/commons/upload/license/MediaLicenseFragment.java app/src/main/java/fr/free/nrw/commons/upload/UploadActivity.java app/src/main/java/fr/free/nrw/commons/upload/UploadMediaDetailAdapter.java app/src/main/java/fr/free/nrw/commons/upload/ThumbnailsAdapter.java app/src/main/java/fr/free/nrw/commons/upload/SimilarImageDialogFragment.java app/src/main/java/fr/free/nrw/commons/upload/mediaDetails/UploadMediaDetailFragment.java app/src/main/java/fr/free/nrw/commons/upload/categories/UploadCategoriesFragment.java app/src/main/java/fr/free/nrw/commons/upload/depicts/DepictsFragment.java

Hi, sorry for not checking it. I was also working towards it, thats why I like getting assigned first. Nevermind, I will close the PR will do the code review on your's PR as I have hands on at the code already.

**Comment by shashankiitbhu** [comment]
> > Hi @nicolas-raoul , following are covered in the Above PRs (Each PR addresses 1 Package, except the last PR) :
> > app/src/main/java/fr/free/nrw/commons/category/CategoryDetailsActivity.java app/src/main/java/fr/free/nrw/commons/settings/SettingsActivity.java
> > app/src/main/java/fr/free/nrw/commons/contributions/MainActivity.java app/src/main/java/fr/free/nrw/commons/contributions/ContributionsFragment.java app/src/main/java/fr/free/nrw/commons/contributions/ContributionViewHolder.java
> > app/src/main/java/fr/free/nrw/commons/explore/ExploreListRootFragment.java app/src/main/java/fr/free/nrw/commons/explore/ExploreFragment.java app/src/main/java/fr/free/nrw/commons/explore/depictions/WikidataItemDetailsActivity.java app/src/main/java/fr/free/nrw/commons/explore/map/ExploreMapFragment.java app/src/main/java/fr/free/nrw/commons/explore/ExploreMapRootFragment.java app/src/main/java/fr/free/nrw/commons/explore/recentsearches/RecentSearchesFragment.java app/src/main/java/fr/free/nrw/commons/explore/SearchActivity.java
> > app/src/main/java/fr/free/nrw/commons/bookmarks/pictures/BookmarkPicturesFragment.java app/src/main/java/fr/free/nrw/commons/bookmarks/BookmarkFragment.java app/src/main/java/fr/free/nrw/commons/bookmarks/BookmarkListRootFragment.java app/src/main/java/fr/free/nrw/commons/bookmarks/locations/BookmarkLocationsFragment.java app/src/main/java/fr/free/nrw/commons/bookmarks/items/BookmarkItemsFragment.java
> > app/src/main/java/fr/free/nrw/commons/profile/leaderboard/LeaderboardFragment.java app/src/main/java/fr/free/nrw/commons/profile/ProfileActivity.java
> > app/src/main/java/fr/free/nrw/commons/upload/license/MediaLicenseFragment.java app/src/main/java/fr/free/nrw/commons/upload/UploadActivity.java app/src/main/java/fr/free/nrw/commons/upload/UploadMediaDetailAdapter.java app/src/main/java/fr/free/nrw/commons/upload/ThumbnailsAdapter.java app/src/main/java/fr/free/nrw/commons/upload/SimilarImageDialogFragment.java app/src/main/java/fr/free/nrw/commons/upload/mediaDetails/UploadMediaDetailFragment.java app/src/main/java/fr/free/nrw/commons/upload/categories/UploadCategoriesFragment.java app/src/main/java/fr/free/nrw/commons/upload/depicts/DepictsFragment.java
> 
> Hi, sorry for not checking it. I was also working towards it, thats why I like getting assigned first. Nevermind, I will close the PR will do the code review on your's PR as I have hands on at the code already.

No issues, also this issue is of such nature that it can't be assigned to one person (as discussed by mentors above) and Ideal way is to contribute 1 PR per package at our pace. Our goal here is to complete this task as a team, thanks and good luck

**Comment by kanahia1** [comment]
Hey @shashankiitbhu, Can I also work on it ?

**Comment by shashankiitbhu** [comment]
@kanahia1 Sure, I am already working on media package (almost done with it) so you can start on others which are left like campaigns

**Comment by neeldoshii** [comment]
>app/src/main/java/fr/free/nrw/commons/notification/NotificationActivity.java
>app/src/main/java/fr/free/nrw/commons/nearby/fragments/NearbyParentFragment.java
>app/src/main/java/fr/free/nrw/commons/campaigns/CampaignView.java

Taking it up

**Comment by kanahia1** [comment]
Hey @neeldoshii, are you working on them ?

**Comment by shashankiitbhu** [comment]
@kanahia1 @neeldoshii Also If you get stuck somewhere (specially if your tests are failing) then take a look at one of my PRs (above PR for Review Package is something you can take reference from), it will save you a lot of time and trouble and do let me know if you need help somewhere

**Comment by neeldoshii** [comment]
> Hey @neeldoshii, are you working on them ?

yes.

**Comment by neeldoshii** [comment]
Hey @psh, I don't think this issue is completed. There are few packages left for this to complete this issue. 
#5606 #5604

**Comment by shashankiitbhu** [comment]
@nicolas-raoul as this task is completed (only some OPEN PRs need to be merged) , shouldn't this be removed from the Phabricator Proposal Page?

**Comment by psh** [comment]
Great work on this migration effort!  I did a search in the code and it turned up a couple of cleanup steps still left to do.

Looks like there is still a little migration to go (binding of views, and click listeners)
* UploadActivity (`@OnClick` was missed in previous conversion)
* NearbyParentFragment (lots of `@BindView` and `@OnClick`)

Cleanup unused butterknife imports in
* MediaLicenseFragment
* BookmarkLocationsFragment
* UserDetailAdapter

Once those are done, we should be able to remove `BUTTERKNIFE_VERSION` from gradle properties, and butterknife itself from build.gradle (both kapt and implementation dependency)

**Comment by shashankiitbhu** [comment]
@psh `NearbyParentFragment` is migrated completely in #5655

**Comment by shashankiitbhu** [comment]
> Cleanup unused butterknife imports in
> 
> * MediaLicenseFragment
> * BookmarkLocationsFragment
> * UserDetailAdapter
> 
> Once those are done, we should be able to remove `BUTTERKNIFE_VERSION` from gradle properties, and butterknife itself from build.gradle (both kapt and implementation dependency)

On it , will be sharing a PR soon

**Comment by neeldoshii** [comment]
> MediaLicenseFragment

Can I take this one @shashankiitbhu please?

**Comment by shashankiitbhu** [comment]
@psh My PR above removes butterknife and completes migration.

**Comment by shashankiitbhu** [comment]
@neeldoshii your PR removed unused Imports from BookmarkLocationsFragment and MediaLicenceFragment, well no worries, My PR is focused on removing other Butterknife elements, you should have confirmed first but no issues.

**Comment by shashankiitbhu** [comment]
@psh I think #5655 have to merged first to complete this migration , can you help in reviewing that ?

**Comment by nicolas-raoul** [comment]
Sorry for the delay, I merged https://github.com/commons-app/apps-android-commons/pull/5655 :-)
